src/server/game/Entities/Object/Object.cpp

```diff
@@ -1147,7 +1147,12 @@ bool WorldObject::IsWithinLOSInMap(const WorldObject* obj) const
         return false;
 
     float ox, oy, oz;
-    obj->GetPosition(ox, oy, oz);
+    float angle = obj->GetAngle(GetPositionX(), GetPositionY());
+    float range = std::min(GetObjectSize(), GetExactDist(obj->GetPositionX(), obj->GetPositionY(), obj->GetPositionZ()));
```

Comment:
Well the purple line is what I think it would have done using the GetObjectSize rather than obj->GetObjectSize. e.g. it would use the size of the from object as bounds for the to object.. Which would be a much bigger problem for example if the from object was much bigger than the "to" object.. :)

As for your problem.. Maybe put a TC_LOG_INFO with the results of the checks for both calls and see if anything stands out as odd?